privacyidea/static/components/token/views/token.details.html

```diff
@@ -362,6 +362,22 @@ <h3 name="Token Details"><translate>Token details for {{ tokenSerial }}
             </td>
         </tr>
 
+        <tr ng-hide="token.locked || !checkRight('setrandompin')">
```

Comment:
I'm not entirely sure about the UI here:
* I think having an extra row that only contains the "set random PIN" and, after a button click, the randomly generated PIN looks a bit weird. What do you think about moving the "set random PIN" button next to the ordinary "set PIN" button, and displaying the randomly generated PIN in a modal dialog with a message "The randomly generated PIN is:" instead? 
* The endpoint ``/token/setrandompin`` does not work without a OTPPINRANDOM policy. But the "set random PIN" button is displayed in any case, which I find it unfortunate. What do you think? We could, for example, use a default length to make the endpont work even without a OTPPINRANDOM policy.

Reply:
I am also not completely happy with that. But I was not able to come up with better ideas. Here are my thoughts:

I personally do not like modal dialogs! ;-)
I think we will often use the "set random pin" with an event handler, that actually removes the PIN from the response. So we will get an empty modal dialog. This is wired. We get the blue infogox that the pin is set. I think this is fine.

I think an additional line is not the problem. Because either policies will be configured this way, that a helpdesk user is allowed to manually set a pin or to set a random pin.

In regards to the default length: I will give it some thinking...
This would also avoid raising the error above. But currently I can not remember if there was a reason for that. 

In near future I am sure, that we not only would want to define the length of the random PIN but also the charset (like only set digits)

Reply:
Hm, I understand your hesitation regarding the modal dialog -- But in the default configuration (without any policies), the token view now contains an additional row only with a "Set random PIN", which in my opinion looks a bit weird :) And once a corresponding ``otp_pin_set_random`` is defined, I find it kind of counter-intuitive that a clicking "Set random PIN" makes the PIN appear in the second cell of the row, while the first cell is still empty. :-/ Maybe we could add a label like
```
<span ng-show="randomPin">Randomly generated PIN:</span>
```
to the first cell of the row?

Reply:
I make a translateable "random generated PIN".

I am not sure if randomly generated means it might be generated or not.
But it means if it is generated, it is a random PIN ;-)

Reply:
done in 8107fb3d

Reply:
thx :)